package javaexp.a10_exception;

public class A01_Basic {

	public static void main(String[] args) {
		// 1. 예외 처리 기본 1단계 처리
		System.out.println("line1");
		System.out.println("line2");
		try { // 예외 처리 블럭으로 만약 처리하지 않으면 에러 발생과 동시에 해당 라인에서 진행 중단
			  // 하지만 try{} catch 문으로 인해 계속 수행 가능
			System.out.println("--- 코드 블럭 ---");
			System.out.println("line3");
			System.out.println("line4");
			// ArrayIndexOutOfBoundsException, 예외 발생. 아래 line5 실행X
			System.out.println(args[0]); 
			System.out.println("line5");
			System.out.println("line6");
			
		} catch(Exception e) { // 예외 처리가 없을 시엔 실행 X
			System.out.println("--- 예외 처리 블럭 ---");
			System.out.println("line7");
			System.out.println("line8");
		}
		
		System.out.println("--- 예외 처리 후 블럭 ---");
		System.out.println("line9");

	}

}
